Project.toml

```diff
@@ -4,7 +4,9 @@ authors = ["David A. van Leeuwen <david.vanleeuwen@gmail.com>"]
 version = "0.9.4"
 
 [deps]
+CSV = "336ed68f-0bac-5ca0-87d4-7b16caf5d00b"
 Combinatorics = "861a8166-3701-5b0c-9a16-15d98fcdc6aa"
+DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
```

Comment:
DataFrames is a quite heavy dependency to take for NamedArrays. Better use the Tables.jl interface instead. You won't probably won't be able to define `convert` methods then, but anyway the fact that you add keyword arguments to `convert` sounds like an indication that it would better be a separate function.

Reply:
Hey @nalimilan,

Thanks for your review.

I wasn't aware of the of the Tables.jl package. I can look to rewrite it to use Tables.jl if desired? But a quick Github search seems to indicate [DataFrames](https://github.com/search?q=using+Dataframes+language%3Ajulia) is used more often (than [Tables.jl](https://github.com/search?q=using+Tables+language%3Ajulia)). Still if you find the dependency to heavy I can always just place the code in a gist somewhere for people to find (although I guess it seems this would be the most logical place to find such a conversion?)

Furthermore the type conversion works without any **required** keywords. 
e.g.
```julia
cn = convert(DataFrame,n)
ccn = convert(NamedArray,cn)
```

It's just that you need to give a column a name and the name I pick ("Values" by default) might not suite everyone. I'm open to change that too if desired.

Reply:
A long time ago I used a package that would provide for some form of auto-loading when another package was loaded.  I can't find that code anymore, it must have been deprecated.  So the idea was:

```
using NamedArrays ## basic code is loaded
using DataFrames ## this triggers loading of DataFrames support in NamedArrays
```
This way we could get rid of the heavy dependence. 

Another option is to have a separate package NamedArrays.DataFrames, but I don't know how submodules work

Reply:
Tables.jl covers DataFrames and many other packages, so it doesn't make a lot of sense to say that DataFrames is used more often. If you support Tables.jl you support basically any table type that exists in the Julia ecosystem for free.

What I meant about keyword arguments is that I don't think it's common for `convert` methods to support keyword arguments. Often if they are useful then it's worth defining a new function.

Reply:
> A long time ago I used a package that would provide for some form of auto-loading when another package was loaded. I can't find that code anymore, it must have been deprecated. So the idea was:

It's probably Requires.jl. It still exists, but currently it slows down loading the package, which kind of defeats the point of not using a plain dependency. So it's not widely used at the moment.